### PR TITLE
Convert to Go modules

### PR DESCRIPTION
**Pull request by bboreham**
Done by running `go mod init` then fixing compilation errors then `go mod vendor`.
A few C files needed to be manually added.

There are a lot of files moved or deleted, because `go mod vendor` flattens dependencies at the top level and the previous tool, `gvt`, did not.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,9 +23,8 @@ RM=--rm
 RUN_FLAGS=-ti
 BUILD_IN_CONTAINER=true
 GO_ENV=GOGC=off
-GO_BUILD_INSTALL_DEPS=-i
```
Comment (fbarl) [inline Makefile]: Do we want to drop `-i` from the instruction `echo "    sudo go clean -i net";` down below as well?

**Comment by fbarl** [comment]
Thanks for taking your time with this @bboreham!

Good catch with the last commit as well, as you mentioned `bpf.h` and `bpf_map.h` are waiting for https://github.com/iovisor/gobpf/pull/206, but what about `vendor/github.com/weaveworks/tcptracer-bpf/tcptracer-bpf.h`? Shall we open a PR that fixes that one?

Overall, the code looks good, wanted to look into the build and test the behavior quickly! :)

File: Makefile (hunk 1 above)
Comment (bboreham) [reply Makefile]: `-i` is actually correct there, as the intention was to install a different build of the standard library, but the whole bit is irrelevant now that Go caches builds for different tags in different places, so I'll remove it.

**Comment by bboreham** [comment]
I made https://github.com/weaveworks/tcptracer-bpf/pull/69 to avoid the manual copy for one file.

**Pull request merged**